contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2472,6 +2477,11 @@ class tsqlBuilder : public tsqlCommonMutator
 					size_t startPosition = id->keyword()->TRANSLATE()->getSymbol()->getStartIndex();
 					rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair("", "sys.")));
 				}
+				if (id->keyword()->SUBSTRING()) /* SUBSTRING */
+				{
+					size_t startPosition = id->keyword()->SUBSTRING()->getSymbol()->getStartIndex();
+					rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair("", "sys.")));
+				}
```

Comment:
I believe this is not needed in tsqlBuilder because it is inherited from tsqlCommonMutator. Can you explain why we are making same changes at multiple places?

Reply:
`TSqlParserBaseListener` handles rewrite in case of Views and `tsqlBuilder` handles rewrite for other statements, we have already done rewrites in both mutators for previous string functions as well as Geosptatial functions

Reply:
it is redundant, the whole `exitFunction_call` or any other exitcall is only needed in tsqlCommonMutator. And since tsqlBuilder is subclass of tsqlCommonMutator, the function definition will be inherited.

Reply:
if the exitFunction_call was not present in tsqlBuilder class then there won't have been an issue we could have put the change only in tsqlCommonMutator, but since the function exitFunction_call is present in tsqlBuilder, during query execution, tsqlBuilder::exitFunction_call will be use as we are overriding the definition of exitFunctioncall, So that's the reason we have to add the logic to both the mutators.

Reply:
Done